client/components/course/Sidebar/Relationship.vue

```diff
@@ -73,9 +88,14 @@ export default {
     getAssociationIds(activity) {
       return get(activity, `refs.${this.type}`, []);
     },
-    onRelationshipChanged(associations) {
+    onRelationshipChanged(value) {
+      const associations = Array.isArray(value) ? value : [value];
       let activity = cloneDeep(this.activity) || {};
-      set(activity, `refs.${this.type}`, map(associations, 'id'));
+      const ids = associations.reduce((acc, it) => {
+        if (it && it.id) acc.push(it.id);
```

Comment:
Thing is if you deselect in multiple mode it will give you array of nulls. And instead of mapping and filtering we are doing that here in single pass.